substrate/client/network/src/litep2p/mod.rs

```diff
@@ -439,6 +441,9 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkBackend<B, H> for Litep2pNetworkBac
 		let peer_store_handle = params.network_config.peer_store_handle();
 		let executor = Arc::new(Litep2pExecutor { executor: params.executor });
 
+		let limit_discovery_under =
+			params.network_config.network_config.default_peers_set.out_peers as usize + 15;
```

Comment:
What happens if `in_peers` is lower than 15?

Do you think the discovery should be guided by the number of sync peers connected, and not by the number of known peers?

Also, I would make sure we do a random walk at least once per minute as it was before, even if we have enough peers connected (make `discovery_only_if_under_num` slow down queries, but not stop them completely).